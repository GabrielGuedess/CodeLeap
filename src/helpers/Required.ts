export type Required<T> = {
  [P in keyof T]-?: T[P];
};
