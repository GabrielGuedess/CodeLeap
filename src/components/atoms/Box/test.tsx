import { renderWithTheme } from 'utils/tests/helpers';

import { Box } from '.';

describe('<Box />', () => {
  it('should render the box correctly', () => {
    // Arrange
    const { container } = renderWithTheme(
      <Box>
        <span>Content</span>
      </Box>,
    );

    // Assert
    expect(container.firstChild).toMatchSnapshot();
  });
});
