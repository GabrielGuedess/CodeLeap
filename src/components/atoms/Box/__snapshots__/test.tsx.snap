// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should render the box correctly 1`] = `
.c0 {
  overflow: hidden;
  background: #FFFFFF;
  border: 1px solid #CCCCCC;
  border-radius: 1.6rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 2.4rem;
  padding: 2.4rem;
}

<div
  class="c0"
>
  <span>
    Content
  </span>
</div>
`;
